src/p2p/net_node.inl

```diff
@@ -1620,18 +1640,16 @@ namespace nodetool
 
     if (!m_exclusive_peers.empty()) return true;
 
-    // Only have seeds in the public zone right now.
-
-    size_t start_conn_count = get_public_outgoing_connections_count();
-    if(!get_public_white_peers_count() && !connect_to_seed())
+    for(auto& zone : m_network_zones)
     {
-      return false;
-    }
+      size_t start_conn_count = get_outgoing_connections_count(zone.second);
+      if(!zone.second.m_peerlist.get_white_peers_count() && !connect_to_seed(zone.first))
+      {
+        return false;
```

Comment:
Should that be continue instead ? (and maybe return false if all fail) ?

Reply:
Yup, thought about this shortly after I posted. And I think a bit more tracking needs to be done to keep the status code accurate (if all seeds fail for all networks, `return false`).